packages/phone-number-privacy/src/index.ts

```diff
@@ -1,21 +1,55 @@
-import { BLINDBLS } from 'bls12377js-blind'
+import { PhoneNumberUtils } from '@celo/utils'
 import * as functions from 'firebase-functions'
-import config, { connectToDatabase } from './config'
+import { confirmUser } from './common/identity'
+import { incrementQueryCount } from './database/wrappers/account'
+import { computeBLSSalt } from './salt-generation/bls-salt'
+import { getRemainingQueryCount } from './salt-generation/query-quota'
 
-export const getSalt = functions.https.onRequest((request, response) => {
-  const privateKey = new Buffer(config.salt.key)
-  try {
-    // Adding this here as an example of how to connect to the DB
-    const knex = connectToDatabase()
-    knex('accounts')
-      .first()
-      .then((val) => console.debug('account data', val))
-      .catch((reason) => console.error(reason))
+export const NUM_ATTESTATIONS_REQUIRED = 3
 
-    const salt = BLINDBLS.computePRF(privateKey, new Buffer(request.body.blindPhoneNumber))
+export const getSalt = functions.https.onRequest(async (request, response) => {
+  try {
+    if (!isValidInput(request.body)) {
+      response.status(400).send('Invalid input parameters')
+      return
+    }
+    confirmUser()
+    const remainingQueryCount = await getRemainingQueryCount(
+      request.body.account,
+      request.body.phoneNumber
+    )
+    if (remainingQueryCount <= 0) {
+      response.status(400).send('Requester exceeded salt service query quota')
+      return
+    }
+    const salt = computeBLSSalt(request.body.queryPhoneNumber)
+    await incrementQueryCount(request.body.account).catch((error) => {
```

Comment:
Should be no need for a `.catch` here given the outer try/catch in this function

Reply:
I suppose -- I was leaving it open in case we wanted to handle this differently.  right now if the db request fails we allow the query to be made.  if i allow the try/catch to be used it would return a 500.  it depends what we how we want to handle these errors